docs/source/tutorial/neighborloader.rst

```diff
@@ -0,0 +1,169 @@
+
+To use it in prctice, we can take a look at the `Reddit <https://github.com/pyg-team/pytorch_geometric/blob/master/examples/reddit.py>`__ example from PyG:
+
+
+Defining our NeighborLoader
+---------------------------
+
+.. code-block:: python
+
+    train_loader = NeighborLoader(data, input_nodes=data.train_mask,
+                                num_neighbors=[25, 10], shuffle=True, **kwargs)
+
+    subgraph_loader = NeighborLoader(copy.copy(data), input_nodes=None,
+                                    num_neighbors=[-1], shuffle=False, **kwargs)
+
+    # No need to maintain these features during evaluation:
+    del subgraph_loader.data.x, subgraph_loader.data.y
+    # Add global node index information.
+    subgraph_loader.data.num_nodes = data.num_nodes
+    subgraph_loader.data.n_id = torch.arange(data.num_nodes)
+
+Note ``data`` has the ``train_mask``, ``val_mask``, and ``test_mask`` attributes, which we use to define our training and evaluation sets.
+``num_neighbors`` means we define a two-layer GNN, where the first layer samples 25 neighbors and the second layer samples 10 neighbors.
+``input_nodes`` is a list of nodes. We need to compute their embedding.
+``shuffle`` means that we train nodes in a different order for each epoch.
+
+We also create a single-hop evaluation neighbor loader: ``subgraph_loader`` is used for evaluation, where we do not need to maintain the features.
+We also add global node index information to the subgraph loader. The global node index is used for index features in the CPU and sends features to the GPU for computation,
+refer line: ``x = x_all[batch.n_id.to(x_all.device)].to(device)``
+
+Define Model
+--------------
+
+.. code-block:: python
+
+    class SAGE(torch.nn.Module):
+        def __init__(self, in_channels, hidden_channels, out_channels):
+            super().__init__()
+            self.convs = torch.nn.ModuleList()
+            self.convs.append(SAGEConv(in_channels, hidden_channels))
+            self.convs.append(SAGEConv(hidden_channels, out_channels))
+
+        def forward(self, x, edge_index):
+            for i, conv in enumerate(self.convs):
+                x = conv(x, edge_index)
+                if i < len(self.convs) - 1:
+                    x = x.relu_()
+                    x = F.dropout(x, p=0.5, training=self.training)
+            return x
+
+        @torch.no_grad()
+        def inference(self, x_all, subgraph_loader):
+            for i, conv in enumerate(self.convs):
+                xs = []
+                for batch in subgraph_loader:
+                    x = x_all[batch.n_id.to(x_all.device)].to(device)
+                    x = conv(x, batch.edge_index.to(device))
+                    if i < len(self.convs) - 1:
+                        x = x.relu_()
+                    xs.append(x[:batch.batch_size].cpu()) # we only need the representations of the target nodes
+                x_all = torch.cat(xs, dim=0)
+            return x_all
```

Comment:
We would need to go a bit more into depth here if we wanna take about layer-wise computation of node embeddings for inference.

Reply:
Do you mean we need to explain why we need a layer-wise way?

Reply:
> We would need to go a bit more into depth here if we wanna take about layer-wise computation of node embeddings for inference.

It seems out of scope. I would like to remove this part.